torch/testing/_asserts.py

```diff
@@ -157,7 +228,13 @@ def _check_attributes_equal(
     if check_dtype and actual.dtype != expected.dtype:
         return AssertionError(msg_fmtstr.format("dtype", actual.dtype, expected.dtype))
 
-    if check_stride and actual.stride() != expected.stride():
+    if actual.is_sparse != expected.is_sparse:
+        return AssertionError(msg_fmtstr.format("is_sparse", actual.is_sparse, expected.is_sparse))
+    elif actual.is_sparse and check_is_coalesced and actual.is_coalesced() != expected.is_coalesced():
+        return AssertionError(msg_fmtstr.format("is_coalesced()", actual.is_coalesced(), expected.is_coalesced()))
+    elif actual.is_sparse_csr != expected.is_sparse_csr:
```

Comment:
What will happen when a new sparse format is introduced?

Reply:
This shouldn't sound snarky, but we would need a new `elif` clause? As long as we keep adding a new bool flag for sparse formats there is little we can do otherwise. If we had for example the combination of `is_sparse: bool` and  `sparse_format: Literal["COO", "CSR", ...]` we could make this nicer.

Reply:
Makes sense, but what about erroring out if an unexpected sparse format is encountered?

Reply:
How about using the following approach:
```
if actual.is_sparse:
    <do sparse COO checks>
elif actual.is_sparse_csr:
    <do sparse CSR checks>
elif actual.layout == torch.strided:
    <do strided checks>
else:
    raise NotImplementedError(f"{actual.layout}")
```
?

Reply:
I've redesigned the checks to follow the approach I outlined above. It can now be extended quite easily and will also emit expressive messages for strided vs. sparse as well as for mismatching sparse formats.